Azure DevOps Pipeline Example: fatal: ambiguous argument 'master': unknown revision or path not in the working tree.

### Documentation issue

<!-- (Update "[ ]" to "[x]" to check a box) -->

- [ ] Reporting a typo
- [ ] Reporting a documentation bug
- [ ] Documentation improvement
- [ ] Documentation feedback

<!--
  If your issue is not regarding the documentation, please choose an issue type:
  https://github.com/nrwl/nx/issues/new/choose
-->

### Is there a specific documentation page you are reporting?

Enter the URL or documentation section here.
https://nx.dev/recipes/ci/monorepo-ci-azure

### Additional context or description

Provide any additional details here as needed.

The code snippet for the azure-pipelines.yaml file does not work in Devops. As soon as the below step is ran, DevOps throws an error

`- script: npx nx format:check`

> fatal: ambiguous argument 'master': unknown revision or path not in the working tree.

Full Log
```
Starting: CmdLine
==============================================================================
Task         : Command line
Description  : Run a command line script using Bash on Linux and macOS and cmd.exe on Windows
Version      : 2.212.0
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/command-line
==============================================================================
Generating script.
Script contents:
npx nx format:check
========================== Starting Command Output ===========================
/usr/bin/bash --noprofile --norc /home/vsts/work/_temp/79102d64-d806-43d7-ba5b-38f3870d1b15.sh
fatal: ambiguous argument 'master': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
apps/api-e2e/jest.config.ts
apps/api-e2e/src/api/api.spec.ts
apps/api-e2e/src/support/global-setup.ts
apps/api-e2e/src/support/test-setup.ts
apps/api/jest.config.ts
apps/api/src/app/app.controller.spec.ts
apps/api/src/app/app.controller.ts
apps/api/src/app/app.module.ts
apps/api/src/app/app.service.spec.ts
apps/api/src/app/app.service.ts
apps/api/src/main.ts
apps/api/webpack.config.js
apps/app/src/app/app.component.ts
jest.config.ts
jest.preset.js
##[error]Bash exited with code '1'.
Finishing: CmdLine
```

Using npx 9.5.0

If I run `npx nx format:check` locally, it works fine. Are there additional arguments that need to be passed here?

--- a/docs/shared/monorepo-ci-azure.md
+++ b/docs/shared/monorepo-ci-azure.md
@@ -2,9 +2,21 @@
 
 Below is an example of an Azure Pipelines setup for an Nx workspace - building and testing only what is affected.
 
+{% callout type="note" title="Check your Shallow Fetch settings" %}
+
+Nx needs additional Git history available for `affected` to function correctly. Make sure Shallow fetching is disabled in your pipeline settings UI. For more info, check out this article from Microsoft [here](https://learn.microsoft.com/en-us/azure/devops/pipelines/yaml-schema/steps-checkout?view=azure-pipelines#shallow-fetch).
+
+{% /callout %}
+
 Unlike `GitHub Actions` and `CircleCI`, you don't have the metadata to help you track the last successful run on `main`. In the example below, the base is set to `HEAD~1` (for push) or branching point (for pull requests), but a more robust solution would be to tag an SHA in the main job once it succeeds and then use this tag as a base. You can also try [using the devops CLI within the pipeline yaml](/recipes/other/azure-last-successful-commit). See the [nx-tag-successful-ci-run](https://github.com/nrwl/nx-tag-successful-ci-run) and [nx-set-shas](https://github.com/nrwl/nx-set-shas) (version 1 implements tagging mechanism) repositories for more information.
 
-We also have to set `NX_BRANCH` explicitly.
+{% callout type="note" title="Tracking the origin branch" %}
+
+If you're using this action in the context of a branch you may need to add `run: "git branch --track main origin/main"` before running the `nx affected` command since `origin/main` won't exist.
+
+{% /callout %}
+
+We also have to set `NX_BRANCH` explicitly. NX_BRANCH does not impact the functionality of your runs, but does provide a human-readable label to easily identify them in the Nx Cloud app.
 
 ```yaml
 trigger:
@@ -15,12 +27,10 @@ pr:
 variables:
   CI: 'true'
   ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
-    NX_BRANCH: $(System.PullRequest.PullRequestNumber)
+    NX_BRANCH: $(System.PullRequest.PullRequestId) # You can use $(System.PullRequest.PullRequestNumber if your pipeline is triggered by a PR from GitHub ONLY)
     TARGET_BRANCH: $[replace(variables['System.PullRequest.TargetBranch'],'refs/heads/','origin/')]
-    BASE_SHA: $(git merge-base $(TARGET_BRANCH) HEAD)
   ${{ if ne(variables['Build.Reason'], 'PullRequest') }}:
     NX_BRANCH: $(Build.SourceBranchName)
-    BASE_SHA: $(git rev-parse HEAD~1)
   HEAD_SHA: $(git rev-parse HEAD)
 
 jobs:
@@ -28,9 +38,26 @@ jobs:
     pool:
       vmImage: 'ubuntu-latest'
     steps:
+      # Set Azure Devops CLI default settings
+      - bash: az devops configure --defaults organization=$(System.TeamFoundationCollectionUri) project=$(System.TeamProject)
+      displayName: 'Set default Azure DevOps organization and project'
+
+      # Get last successfull commit from Azure Devops CLI
+      - bash: |
+        LAST_SHA=$(az pipelines build list --branch $(Build.SourceBranchName) --definition-ids $(System.DefinitionId) --result succeeded --top 1 --query "[0].triggerInfo.\"ci.sourceSha\"")
+        if [ -z "$LAST_SHA" ]
+        then
+          LAST_SHA=$DEFAULT_BASE_SHA
+        fi
+        echo "Last successful commit SHA: $LAST_SHA"
+        echo "##vso[task.setvariable variable=BASE_SHA]$LAST_SHA"
+      displayName: 'Get last successful commit SHA'
+      env:
+        AZURE_DEVOPS_EXT_PAT: $(System.AccessToken)
+
       - script: npm ci
 
-      - script: npx nx format:check
+      - script: npx nx format:check --base=$(BASE_SHA)
 
       - script: npx nx affected --base=$(BASE_SHA) -t lint --parallel=3
       - script: npx nx affected --base=$(BASE_SHA) -t test --parallel=3 --configuration=ci

